zerver/views/webhooks/updown.py

```diff
@@ -0,0 +1,77 @@
+# Webhooks for external integrations.
+from __future__ import absolute_import
+import re
+from datetime import datetime, timedelta
+from zerver.lib.actions import check_send_message
+from zerver.lib.response import json_success, json_error
+from zerver.decorator import REQ, has_request_variables, api_key_only_webhook_view
+
+
+SUBJECT_TEMPLATE = "{service_url}"
+
+
+class UnsupprotedUpdownEventType(Exception):
+    pass
+
+def send_message_for_event(event, user_profile, client, stream):
+    event_type = get_event_type(event)
+    subject = SUBJECT_TEMPLATE.format(service_url=event['check']['url'])
+    body = EVENT_TYPE_BODY_MAPPER[event_type](event)
+    check_send_message(user_profile, client, 'stream', [stream], subject, body)
+
+def get_body_for_up_event(event):
+    body = "Service is `up`"
+    event_downtime = event['downtime']
+    if event_downtime['started_at']:
+        body = "{} again".format(body)
+        string_date = get_time_string_based_on_duration(event_downtime['duration'])
+        if string_date:
+            body = "{} after {}".format(body, string_date)
+    return "{}.".format(body)
+
+def get_time_string_based_on_duration(duration):
```

Comment:
I feel like there's probably a stdlib function that does this; it'd be better to use that since probably this function is subtly wrong around daylight savings time.